sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -0,0 +1,968 @@
+        if (currentReader != null) {
+          currentReader.close();
+          currentReader = null;
+        }
+        currentRecord = null;
+      }
+
+      @Override
+      public Double getFractionConsumed() {
+        if (doneReading) {
+          return 1.0;
+        }
+        synchronized (currentReader) {
+          if (currentReader == null || recordsReturned == 0) {
+            return 0.0;
+          }
+          return getProgress();
+        }
+      }
+
+      /**
+       * Returns RecordReader's Progress.
+       */
+      private Double getProgress() {
+        try {
+          return (double) currentReader.getProgress();
+        } catch (IOException | InterruptedException e) {
+          LOG.error(HadoopInputFormatIOContants.GETFRACTIONSCONSUMED_ERROR_MSG + e.getMessage(), e);
+          return null;
+        }
+      }
+
+      @Override
+      public final long getSplitPointsRemaining() {
+        if (doneReading) {
+          return 0;
+        }
+        /**
+         * This source does not currently support dynamic work rebalancing, so remaining parallelism
+         * is always 1.
+         */
+        return 1;
+      }
+    }
+
+    /**
+     * A wrapper to allow Hadoop {@link org.apache.hadoop.mapreduce.InputSplit}s to be serialized
+     * using Java's standard serialization mechanisms. Note that the InputSplit has to be Writable
+     * (which mostly are).
+     */
+    public static class SerializableSplit implements Externalizable {
+
+      private static final long serialVersionUID = 0L;
+
+      private InputSplit split;
+
+      public SerializableSplit() {}
+
+      public SerializableSplit(InputSplit split) {
+        checkArgument(split instanceof Writable, String
+            .format(HadoopInputFormatIOContants.SERIALIZABLE_SPLIT_WRITABLE_ERROR_MSG, split));
+        this.split = split;
+      }
+
+      public InputSplit getSplit() {
+        return split;
+      }
+
+      @Override
+      public void writeExternal(ObjectOutput out) throws IOException {
+        out.writeUTF(split.getClass().getCanonicalName());
+        ((Writable) split).write(out);
+      }
+
+      @Override
+      public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+        String className = in.readUTF();
+        try {
+          split = (InputSplit) Class.forName(className).newInstance();
+          ((Writable) split).readFields(in);
+        } catch (InstantiationException | IllegalAccessException e) {
+          throw new IOException("Unable to create split : " + e);
+        }
+      }
+    }
+
+    public boolean producesSortedKeys(PipelineOptions arg0) throws Exception {
+      return false;
+    }
+
+  }
+
+  /**
+   * A wrapper to allow Hadoop {@link org.apache.hadoop.conf.Configuration} to be serialized using
+   * Java's standard serialization mechanisms. Note that the org.apache.hadoop.conf.Configuration
+   * has to be Writable (which mostly are).
```

Comment:
This comment is incorrect. Configuration is always writable. Removed "mostly are"